### PR TITLE
Allow to set up JSON-based wizard information with variables

### PR DESCRIPTION
**Pull request by yarfruct**
This pull request adds possibility to set up the JSON-based wizard information like `displayName`, `displayCategory` and `description` with variables like `"%{VariableName}"`.

### CODE REVIEW

**Comment by vigejolla** [comment]
Please prefix the commit messages with the plugin/library you are modifying e.g. "Sfdk: Add missing includes"

File: src/plugins/mer/merplugin.cpp [hunk 1]
```diff
@@ -154,6 +154,10 @@ bool MerPlugin::initialize(const QStringList &arguments, QString *errorString)
     RunControl::registerWorker<RemoteLinuxQmlProfilerSupport>(QML_PROFILER_RUN_MODE, constraint);
     //RunControl::registerWorker<RemoteLinuxPerfSupport>(PERFPROFILER_RUN_MODE, constraint);
 
+    Utils::globalMacroExpander()->registerVariable("SailfishOSVariant",
+                                                   tr("Category of a wizard for %1 files").arg(Sdk::osVariant()),
+                                                   []() { return Sdk::osVariant(); });
```
Comment (martyone) [inline src/plugins/mer/merplugin.cpp]: I would immediately register all the variables os/ide/sdk variant (similarly to how it is done under sfdk)

File: src/plugins/mer/merplugin.cpp [hunk 2]
```diff
@@ -154,6 +154,16 @@ bool MerPlugin::initialize(const QStringList &arguments, QString *errorString)
     RunControl::registerWorker<RemoteLinuxQmlProfilerSupport>(QML_PROFILER_RUN_MODE, constraint);
     //RunControl::registerWorker<RemoteLinuxPerfSupport>(PERFPROFILER_RUN_MODE, constraint);
 
+    Utils::globalMacroExpander()->registerVariable("SailfishOSVariant",
+                                                   tr("Category of a wizard for %1 files").arg(Sdk::osVariant()),
```
Comment (martyone) [inline src/plugins/mer/merplugin.cpp]: Actually the possible use is not limited for wizard category. So e.g. `"Expands to \"Sailfish OS\" or its variant name"`, `"Expands to \"Sailfish IDE\" or its variant name"`... would be more appropriate.

The variables should be SailfishOsVariant, SailfishIdeVariant and SailfishSdkVariant (notice the case) unless I missed something.

**Comment by yarfruct** [comment]
This PR is closed because now we do not have the opportunity to deal with this task. We will return to her later.

**Pull request closed without merging**